### PR TITLE
Fix compiler warnings in unit tests

### PR DESCRIPTION
**Pull request by akosthekiss**
JerryScript-DCO-1.0-Signed-off-by: Akos Kiss akiss@inf.u-szeged.hu

### CODE REVIEW

**Comment by akosthekiss** [comment]
Note: not all compilers generate warnings during the compilation of unit tests, but OSX clang does. Some examples:
```
tests/unit/test-libm.c:42:49: warning: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
tests/unit/test-api.c:74:55: warning: format specifies type 'void *' but the argument has type 'const jerry_value_t *' (aka 'const unsigned int *') [-Wformat-pedantic]
```

File: tests/unit/test-libm.c [hunk 1]
```diff
@@ -39,7 +39,14 @@ check_double (const char *expr, double computed, double expected)
   uint64_t *computed_bits_p = (uint64_t *) &computed;
   uint64_t *expected_bits_p = (uint64_t *) &expected;
 
-  printf ("%s = 0x%lx [expected=0x%lx] ", expr, *computed_bits_p, *expected_bits_p);
+  uint32_t computed_hi = (uint32_t) (*computed_bits_p >> 32);
+  uint32_t computed_lo = (uint32_t) *computed_bits_p;
+  uint32_t expected_hi = (uint32_t) (*expected_bits_p >> 32);
+  uint32_t expected_lo = (uint32_t) *expected_bits_p;
```
Comment (zherczeg) [inline tests/unit/test-libm.c]: There is a better way to do this with unions. Define a union with a double and an uint32_t [2] array, and do the conversion with it.

Comment (akosthekiss) [inline tests/unit/test-libm.c]: I've rewritten it as suggested. See updated patch. (Note, however, that it's not as big/little-endian-safe. But it's just the printing, so no big deal.)

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**